x/claim/keeper/msg_server_claim_eth.go

```diff
@@ -45,35 +41,47 @@ func (k msgServer) ClaimEth(goCtx context.Context, msg *types.MsgClaimEth) (*typ
 		return nil, errors.New("invalid signature")
 	}
 
+	// create new arkeo claim
+	arkeoClaim := types.ClaimRecord{
+		Address:        msg.Creator,
+		Chain:          types.ARKEO,
+		AmountClaim:    ethClaim.AmountClaim,
+		AmountVote:     ethClaim.AmountVote,
+		AmountDelegate: ethClaim.AmountDelegate,
+	}
+
 	// set eth claim to completed
-	ethClaim.ActionCompleted[types.FOREIGN_CHAIN_ACTION_CLAIM] = true
+	ethClaim = setClaimableAmountForAction(ethClaim, types.ACTION_CLAIM, sdk.Coin{})
```

Comment:
Here is my understanding , correct me if I am wrong, originally during genesis , a claim record will be written to block chain , something like this
```
{
  "address":"0xETH address",
  "chain": types.ETH,
   "amount_claim": 100 * 1e8,
    "amount_vote": 110 * 1e8,
    "amount_delegate": 120 * 1e8
}
```
So here when user first claim , a record will be created on `arkeo` block chain, something like this
```
{
  "address":"arkeo address",
  "chain": types.Arkeo,
   "amount_claim": 100 * 1e8,
    "amount_vote": 110 * 1e8,
    "amount_delegate": 120 * 1e8
}
```
immediately after , the `amount_claim` will be credit to customer's address `arkeo address`   and also `amount_claim` will be set to 0.

However looking at this line, the `claim_amount` of original claim record for `eth` chain will be set to 0
```
{
  "address":"0xETH address",
  "chain": types.ETH,
   "amount_claim": 0,
    "amount_vote": 110 * 1e8,
    "amount_delegate": 120 * 1e8
}
```
1) First of all , this will allow user to claim the same `ETH address` forever 
2) once the first claim comes in , and arkeo record get created , the original claim either should be deleted or all the amount should be set to `0`

Is that make sense?

Reply:
right, but can you please set all amount to zero?

Reply:
yup, will do!